core/src/util/tileID.h

```diff
@@ -16,6 +16,7 @@ struct TileID {
     
     TileID(int _x, int _y, int _z) : x(_x), y(_y), z(_z) {};
     TileID(const TileID& _rhs): x(_rhs.x), y(_rhs.y), z(_rhs.z) {};
+    TileID(const TileID&& _rhs): x(std::move(_rhs.x)), y(std::move(_rhs.y)), z(std::move(_rhs.z)) {};
```

Comment:
I would hesitate to use move on TileIDs everywhere; a move is only appropriate if we don't ever need the original object after that point (kind of defeats the point of trying to make them immutable). Objects this small are trivially simple to copy anyway.